lib/events.js

```diff
@@ -1,75 +1,59 @@
-var events = require('events')
-var util = require('util')
+'use strict'
 
-var helper = require('./helper')
+const EventEmitter = require('events').EventEmitter
+const helper = require('./helper')
 
-var bindAllEvents = function (object, context) {
-  context = context || this
+function bufferEvents (emitter, eventsToBuffer) {
+  const listeners = []
+  const eventsToReply = []
 
-  var bindMethod = function (method) {
-    context.on(helper.camelToSnake(method.substr(2)), function () {
-      var args = Array.prototype.slice.call(arguments, 0)
-      args.push(context)
-      object[method].apply(object, args)
-    })
+  function genericListener (...args) {
+    eventsToReply.push(args)
   }
 
-  for (var method in object) {
-    if (helper.isFunction(object[method]) && method.substr(0, 2) === 'on') {
-      bindMethod(method)
-    }
-  }
-}
-
-var bufferEvents = function (emitter, eventsToBuffer) {
-  var listeners = []
-  var eventsToReply = []
-  var genericListener = function () {
-    eventsToReply.push(Array.prototype.slice.call(arguments))
-  }
-
-  eventsToBuffer.forEach(function (eventName) {
-    var listener = genericListener.bind(null, eventName)
+  eventsToBuffer.forEach(eventName => {
```

Comment:
We seem to have -- mostly -- been using parens even for the one-argument case:
 `eventsToBuffer.forEach((eventName) => {`
This form is more instantly readable as a function and makes adding/removing args easy.

(A couple of places)

Reply:
Ok, I understand and will improve it. 

What do you think about update eslint configuration with [arrow-parens](https://eslint.org/docs/rules/arrow-parens) rule to catch such cases? I can do pr for that.

Reply:
sounds great!